omod/src/main/webapp/dictionary/concept.jsp

```diff
@@ -224,13 +224,16 @@
 		<c:if test="${command.concept.datatype != null && command.concept.datatype.name == 'Coded'}">
 			<tr class="localeSpecific">
 				<th valign="top"><openmrs:message code="Concept.answers"/></th>
-                <c:forEach items="${command.locales}" var="loc">
-                    <td class="${loc}">
-                        <c:forEach items="${command.conceptAnswersByLocale[loc]}" var="answer">
-                            <a href="concept.htm?conceptId=${fn:substring(answer.key, 0, fn:indexOf(answer.key, '^'))}"><c:out value="${answer.value}" /> (${fn:substring(answer.key, 0, fn:indexOf(answer.key, '^'))})</a><br/>
-                        </c:forEach>
-                    </td>
-                </c:forEach>
+				<c:forEach items="${command.locales}" var="locale">
+					<td class="${fn:escapeXml(locale)}">
+						<c:forEach items="${command.conceptAnswersByLocale[locale]}" var="conceptAnswer">
+							<c:set var="conceptId" value="${fn:substring(conceptAnswer.key, 0, fn:indexOf(conceptAnswer.key, '^'))}" />
+							<a href="concept.htm?conceptId=${fn:escapeXml(conceptId)}">
```

Comment:
Will the conceptId ever have a value that needs to be escaped?

Reply:
Yes , For example, if `conceptId` was `1<2`, it would be escaped to `1&lt;2`, which prevents the browser from interpreting it as HTML. This is a common practice when inserting user-provided data into HTML to prevent XSS attacks.

Reply:
conceptId is an integer

Reply:
Yes i understand that , It's always better to be safe than sorry when it comes to preventing potential security vulnerabilities, this will not change the type from interger